### DISCUSSION TITLE
Bisq Onboarding Offers / Bitcoin Meetup Offers / Private Offers

### DISCUSSION THREAD

**Discussion by ghost** [bisq-network/bisq#6215] [Ideas]
**Problem:** A market maker has a group of regular buyers that he trusts and would like to make offers available to them at preferential rates.  

**Problem:** An organizer of a local Bitcoin meetup / onboarding session wants to create some special offers to make it easy to onboard new users to Bisq (at an attractive price for example).  Those offers are likely to get snapped up quickly by other market participants.

**Solution:** The maker flags their offers as "private", associating them with a secret phrase (for example "North Dakota Bitcoin User Group").  Those offers are only visible to & can only be taken by Bisq users that have the secret phrase entered in the UI.

**Implementation:**
- a ui in settings that lets users manage their list of "private offer secret phrases" (used by both offer makers and takers)
- a ui at offer creation to let the maker designate the offer to be "private", and choose the secret phrase for the offer.
- market view : private offers will only be displayed if the user phrase matches the challenge hash within the offer payload.
- taking an offer: the taker sends the challenge phrase in the OfferAvailabilityRequest and InputsForDepositTxRequest messages which the maker validates.

**Comment by chimp1984** [comment]
There are some preparations for that use case in the Offer class. But it was never used and not sure if it would cover all..

**Comment by thoeneberg** [comment]
The community is currently growing enormously. Meetups are springing up like mushrooms.
Personally, I believe that such a function will attract more and more interest.

**Reply by HenrikJannsen** [reply]
A very easy solution is to use the Face to Face payment method and use the exact location. Hardly will get taken from anyone outside the meetup...

**Reply by propropropro22** [reply]
I think software makes it easier to facilitate, and provides security physical and in the form of escrow. Also it helps to flock more people to the platform

**Reply by pazza83** [reply]
Would also be good to have this feature for the future when traders are looking to be more cautious with who they trade with.
